### PR TITLE
Handles error :  not enough space

### PR DESCRIPTION
**Pull request by jeremco**
Hi

This PR handles not enough space error. 
When this error occurs, download is stopped, and there is no retry of chunk download.
The idea is to notify user of no space left, so that he is able to delete some files and resume download. 
He has not to wait for all chunk to be in error, to be notify of the error

Jérémie

### CODE REVIEW

File: api/downloads/download-file-util.js [hunk 1]
```diff
@@ -24,6 +24,7 @@ const errors = {
   CHUNK_SIZE_ERROR: "CHUNK_SIZE_ERROR",
   FILE_CREATING_ERROR: "FILE_CREATING_ERROR",
   FILE_WRITING_ERROR: "FILE_WRITING_ERROR",
+  NO_SPACE_LEFT_ERROR: "ENOSPC",
```
Comment (mackode) [inline api/downloads/download-file-util.js]: Please use same convention like in the other errors for value 'NO_SPACE_LEFT_ERROR'.

Comment (jeremco) [reply api/downloads/download-file-util.js]: ENOSPC is the offical node.js error. It is not an error from downstream. Using the 'convention', I would have to add a transposition function, which is not necessary and efficient

Comment (mackode) [reply api/downloads/download-file-util.js]: But you already do this here: https://github.com/castlabs/downstream_electron/pull/54/files#diff-9e4fcd4cb87f2c825ad90e1fa50cfdbfR87

Wouldn't it be just a matter of changing 
NO_SPACE_LEFT_ERROR: "ENOSPC"

to

NO_SPACE_LEFT_ERROR: "NO_SPACE_LEFT_ERROR",

and have more readable downstream error?

Comment (jeremco) [reply api/downloads/download-file-util.js]: Ok, but then il will have to change in my application
It is just a value. ENOSPC is an official error.

**Comment by jeremco** [comment]
I have changed NO_SPACE_LEFT_ERROR: "ENOSPC" to NO_SPACE_LEFT_ERROR: "NO_SPACE_LEFT_ERROR" in last commit

**Pull request merged**